flyteidl/protos/flyteidl/core/artifact_id.proto

```diff
@@ -19,8 +20,13 @@ message ArtifactBindingData {
   uint32 index = 1;
 
   // These two fields are only relevant in the partition value case
-  string partition_key = 2;
-  string transform = 3;
+  oneof partition_data {
```

Comment:
above this, `ArtifactBindingData` is listed as being only valid for triggers, why is that?